internal/cli/removedb.go

```diff
@@ -0,0 +1,145 @@
+package cli
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/internal/cli/flagset"
+	"github.com/ethereum/go-ethereum/internal/cli/server"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/node"
+
+	"github.com/mitchellh/cli"
+)
+
+// RemoveDBCommand is for removing blockchain and state databases
+type RemoveDBCommand struct {
+	*Meta2
+
+	datadir string
+}
+
+// MarkDown implements cli.MarkDown interface
+func (c *RemoveDBCommand) MarkDown() string {
+	items := []string{
+		"# RemoveDB",
+		"The ```bor removedb``` command will remove the blockchain and state databases at the given datadir location",
+		c.Flags().MarkDown(),
+	}
+
+	return strings.Join(items, "\n\n")
+}
+
+// Help implements the cli.Command interface
+func (c *RemoveDBCommand) Help() string {
+	return `Usage: bor removedb <datadir>
+
+  This command will remove the blockchain and state databases at the given datadir location`
+}
+
+// Synopsis implements the cli.Command interface
+func (c *RemoveDBCommand) Synopsis() string {
+	return "Remove blockchain and state databases"
+}
+
+func (c *RemoveDBCommand) Flags() *flagset.Flagset {
+	flags := c.NewFlagSet("removedb")
+
+	flags.StringFlag(&flagset.StringFlag{
+		Name:  "datadir",
+		Value: &c.datadir,
+		Usage: "Path of the data directory to store information",
+	})
+
+	return flags
+}
+
+// Run implements the cli.Command interface
+func (c *RemoveDBCommand) Run(args []string) int {
+	flags := c.Flags()
+
+	// parse datadir
+	if err := flags.Parse(args); err != nil {
+		c.UI.Error(err.Error())
+		return 1
+	}
+
+	datadir := c.datadir
+	if datadir == "" {
+		datadir = server.DefaultDataDir()
+	}
+
+	// create ethereum node config with just the datadir
+	nodeCfg := &node.Config{DataDir: datadir}
+
+	// Remove the full node state database
+	path := nodeCfg.ResolvePath("chaindata")
+	if common.FileExist(path) {
+		confirmAndRemoveDB(c.UI, path, "full node state database")
+	} else {
+		log.Info("Full node state database missing", "path", path)
```

Comment:
we'd like to continue execution in that case?

Reply:
Resolving this as per our last conversation.